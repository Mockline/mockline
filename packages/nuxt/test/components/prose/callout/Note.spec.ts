import { describe, it, expect } from 'vitest'
import type {TypeOf} from "zod";
// @ts-ignore
import { MNote } from '#components'
import ComponentRender from "../../../component-render";

describe('Note', () => {
  it.each([
    ['basic case', { props: {} }],
    ['<MNote>Note slot</MNote>'],
    // @ts-expect-error
  ])('renders %s correctly', async (nameOrHtml, options: TypeOf<typeof MNote.props>) => {
    const html = await ComponentRender(nameOrHtml, options, MNote)
    expect(html).toMatchSnapshot()
  })
})

