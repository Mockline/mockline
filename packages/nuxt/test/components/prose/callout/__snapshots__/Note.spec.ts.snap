// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Note > renders <MNote>Note slot</MNote> correctly 1`] = `
"<div class="group border relative block rounded-base px-4 py-3 text-sm/6 my-5 dark:border-blue-500/20 dark:bg-blue-800/5 border-blue-500/30 bg-blue-500/10 text-blue-600 transition-colors duration-200 ease-in-out">
  <!--v-if--><span class="iconify i-lucide:info size-4 shrink-0 align-middle mr-1.5 text-blue-600" aria-hidden="true"></span>
  <!--v-if-->Note slot
</div>"
`;

exports[`Note > renders basic case correctly 1`] = `
"<div class="group border relative block rounded-base px-4 py-3 text-sm/6 my-5 dark:border-blue-500/20 dark:bg-blue-800/5 border-blue-500/30 bg-blue-500/10 text-blue-600 transition-colors duration-200 ease-in-out">
  <!--v-if--><span class="iconify i-lucide:info size-4 shrink-0 align-middle mr-1.5 text-blue-600" aria-hidden="true"></span>
  <!--v-if-->
</div>"
`;
